### DISCUSSION TITLE
more command line interface or api wanted [FEAT]

### DISCUSSION THREAD

**Discussion by youzark** [JaDogg/pydoro#268] [Ideas]
**Related Problem**
I am trying to integrate this with taskwarrior timewarrior and vim to build my task system 
I want to trigger the start pause and reset with some command so that I can bind it with taskwarrior and other tools to keep track
of task and generate reports.
**Wanted Features**
It would be nice if you can provide some thing like:
$ pydoro start [jobid]
$ pydoro pause [jobid]
$ pydoro reset [jobid]
-- job id used to refer to the working instance
It would be even better If it can communicate with taskwarrior's data base directly
**Alternative**
or can you provide some functions and document that you think can get this done.
**By the way**
This is the best pomodoro timer I could find in github, I love your style really.

**Comment by JaDogg** [comment]
**Design:**

- When pydoro is executed with `--server` -> show pydoro normally and spawn a background thread that monitors a port configured in .ini config file. (This can be an HTTP server or a simple TCP server, even 1 byte communication should be enough to do start, stop, pause )
   - In that background thread add command to a queue if received.. queue can be read by the UI thread and perform
commands as it were received manually.
- When pydoro [start|stop|reset] is called send a message to configured HTTP/TCP server port..

I'm not sure about `[jobid]` I think we can introduce a `--port` feature if multiple pydoro instances are needed (not sure how that would work as all would go screaming in the ear😆
